src/store/agents.js

```diff
@@ -34,6 +34,18 @@ export const actions = {
 			thought,
 		};
 	},
+	setAgentState: ( state ) => {
```

Comment:
This should not be necessary as long as the Agent's state is a pure function of other state in the system. I'm slightly suspicious of storing nondescript blobs of values in the store, since it creates an opening for bugs. What will you be storing in this value?

Reply:
I created this as a generalized action to be able to update any value in the store. I'm not sure if there would be a good way to dispatch the individual actions from the toolkit callbacks?